rskj-core/src/main/java/co/rsk/peg/union/UnionBridgeSupportImpl.java

```diff
@@ -281,19 +312,52 @@ public UnionResponseCode setTransferPermissions(Transaction tx, boolean requestE
             return UnionResponseCode.SUCCESS;
         }
 
-        storageProvider.setUnionBridgeRequestEnabled(requestEnabled);
-        storageProvider.setUnionBridgeReleaseEnabled(releaseEnabled);
+        RskAddress txSender = tx.getSender(signatureCache);
+        ABICallElection setTransferPermissionsElection = storageProvider.getTransferPermissionsElection(
+            authorizer);
+        ABICallSpec setTransferPermissionsVote = new ABICallSpec(SET_TRANSFER_PERMISSIONS_TAG, new byte[][]{
+            BridgeSerializationUtils.serializeBoolean(requestEnabled),
+            BridgeSerializationUtils.serializeBoolean(releaseEnabled)
+        });
+
+        boolean successfulVote = setTransferPermissionsElection.vote(setTransferPermissionsVote, txSender);
+        if (!successfulVote) {
+            logger.warn("[{}] Unsuccessful {} vote", SET_TRANSFER_PERMISSIONS_TAG, setTransferPermissionsVote);
+            return UnionResponseCode.GENERIC_ERROR;
+        }
 
-        RskAddress caller = tx.getSender(signatureCache);
+        Optional<ABICallSpec> electionWinner = setTransferPermissionsElection.getWinner();
+        if (electionWinner.isEmpty()) {
+            logger.info("[{}}] Successful {} vote.", SET_TRANSFER_PERMISSIONS_TAG, setTransferPermissionsVote);
+            return UnionResponseCode.SUCCESS;
+        }
+
+        ABICallSpec winner = electionWinner.get();
+        Boolean winnerRequestEnabled;
+        Boolean winnerReleaseEnabled;
+        try {
+            winnerRequestEnabled = BridgeSerializationUtils.deserializeBoolean(
```

Comment:
Could we test when there are 2 votes where 1 value matches and 1 differs?

For example, both vote releaseEnabled=true but on votes requestEnabled=true and the other votes requestEnabled=false